source/qb64pe.bas

```diff
@@ -24637,22 +24700,23 @@ FUNCTION EvaluateNumbers$ (p, num() AS STRING)
             'Since we're not using the 2nd number here, we need to return it back so it can still be processed properly
             SELECT CASE OName(p) 'Depending on our operator..
-                CASE "C_UOF": n1~%& = VAL(num(1)): EvaluateNumbers$ = RTRIM$(LTRIM$(STR$(n1~%&))) + num(2)
-                CASE "C_ULO": n1%& = VAL(num(1)): EvaluateNumbers$ = RTRIM$(LTRIM$(STR$(n1%&))) + num(2)
-                CASE "C_UBY": n1~%% = VAL(num(1)): EvaluateNumbers$ = RTRIM$(LTRIM$(STR$(n1~%%))) + num(2)
-                CASE "C_UIN": n1~% = VAL(num(1)): EvaluateNumbers$ = RTRIM$(LTRIM$(STR$(n1~%))) + num(2)
-                CASE "C_BY": n1%% = VAL(num(1)): EvaluateNumbers$ = RTRIM$(LTRIM$(STR$(n1%%))) + num(2)
-                CASE "C_IN": n1% = VAL(num(1)): EvaluateNumbers$ = RTRIM$(LTRIM$(STR$(n1%))) + num(2)
-                CASE "C_UIF": n1~&& = VAL(num(1)): EvaluateNumbers$ = RTRIM$(LTRIM$(STR$(n1~&&))) + num(2)
-                CASE "C_OF": n1~& = VAL(num(1)): EvaluateNumbers$ = RTRIM$(LTRIM$(STR$(n1~&))) + num(2)
-                CASE "C_IF": n1&& = VAL(num(1)): EvaluateNumbers$ = RTRIM$(LTRIM$(STR$(n1&&))) + num(2)
-                CASE "C_LO": n1& = VAL(num(1)): EvaluateNumbers$ = RTRIM$(LTRIM$(STR$(n1&))) + num(2)
-                CASE "C_UBI": n1~` = VAL(num(1)): EvaluateNumbers$ = RTRIM$(LTRIM$(STR$(n1~`))) + num(2)
-                CASE "C_BI": n1` = VAL(num(1)): EvaluateNumbers$ = RTRIM$(LTRIM$(STR$(n1`))) + num(2)
-                CASE "C_FL": n1## = VAL(num(1)): EvaluateNumbers$ = RTRIM$(LTRIM$(STR$(n1##))) + num(2)
-                CASE "C_DO": n1# = VAL(num(1)): EvaluateNumbers$ = RTRIM$(LTRIM$(STR$(n1#))) + num(2)
-                CASE "C_SI": n1! = VAL(num(1)): EvaluateNumbers$ = RTRIM$(LTRIM$(STR$(n1!))) + num(2)
+                CASE "C_UOF": n1~%& = VAL(num(1)): temp$ = STR$(n1~%&)
+                CASE "C_ULO": n1%& = VAL(num(1)): temp$ = STR$(n1%&)
+                CASE "C_UBY": n1~%% = VAL(num(1)): temp$ = STR$(n1~%%)
+                CASE "C_UIN": n1~% = VAL(num(1)): temp$ = STR$(n1~%)
+                CASE "C_BY": n1%% = VAL(num(1)): temp$ = STR$(n1%%)
+                CASE "C_IN": n1% = VAL(num(1)): temp$ = STR$(n1%)
+                CASE "C_UIF": n1~&& = VAL(num(1)): temp$ = STR$(n1~&&)
+                CASE "C_OF": n1~& = VAL(num(1)): temp$ = STR$(n1~&)
+                CASE "C_IF": n1&& = VAL(num(1)): temp$ = STR$(n1&&)
+                CASE "C_LO": n1& = VAL(num(1)): temp$ = STR$(n1&)
+                CASE "C_UBI": n1~` = VAL(num(1)): temp$ = STR$(n1~`)
+                CASE "C_BI": n1` = VAL(num(1)): temp$ = STR$(n1`)
+                CASE "C_FL": n1## = VAL(num(1)): temp$ = STR$(n1##)
+                CASE "C_DO": n1# = VAL(num(1)): temp$ = STR$(n1#)
+                CASE "C_SI": n1! = VAL(num(1)): temp$ = STR$(n1!)
             END SELECT
+            EvaluateNumbers$ = _TRIM$(temp$) + num(2)
             EXIT FUNCTION
         CASE 10 'functions
             SELECT CASE OName(p) 'Depending on our operator..
```

Comment:
Like the one I mentioned at `24682`, I think this select will have the same problem, it doesn't check for the non-underscore names.

Reply:
Good catch.  I meant to fix that and then just got myself sidetracked chasing down a different glitch.  These should now check both for us, properly.